plasmapy/physics/quantum.py

```diff
@@ -281,3 +284,157 @@ def Thomas_Fermi_length(n_e):
     energy_F = Fermi_energy(n_e)
     lambda_TF = np.sqrt(2 * eps0 * energy_F / (3 * n_e * e ** 2))
     return lambda_TF.to(units.m)
+
+
+@check_quantity({
+    'n_e': {'units': units.m**-3, 'can_be_negative': False}
+})
+def Wigner_Seitz_radius(n):
+    r"""Calculate the Wigner-Seitz radius, which approximates the inter-
+    particle spacing. It is the radius of a sphere whose volume is 
+    equal to the mean volume per atom in a solid. This parameter is
+    often used to calculate the coupling parameter.
+    When ion density is used, this is the ion sphere radius.
+
+    Parameters
+    ----------
+    n: Quantity
+        Particle number density
+
+    Returns
+    -------
+    radius: Quantity
+        The Wigner-Seitz radius in meters
+
+    Raises
+    ------
+    TypeError
+        If argument is not a Quantity
+
+    UnitConversionError
+        If argument is in incorrect units
+
+    ValueError
+        If argument contains invalid values
+
+    UserWarning
+        If units are not provided and SI units are assumed
+
+    Notes
+    -----
+    The Wigner-Seitz radius approximates the interparticle spacing. 
+    It is the radius of a sphere whose volume is equal to the mean 
+    volume per atom in a solid:
+
+    .. math::
+    r = \left(\frac{3}{4 \pi n}\right)^{1/3}
+
+    See also
+    --------
+    Fermi_energy
+
+    Example
+    -------
+    >>> from astropy import units as u
+    >>> Wigner_Seitz_radius(1e29 * u.m**-3)
+    <Quantity 1.3365046175719775e-10 m>
+
+    """
+    radius = (3 / (4 * np.pi * n)) ** (1 / 3)
+    return radius
+
+
+def chemical_potential(n_e, T, tol=1e-6):
+    r"""Calculate the ideal chemical potential
+
+    Parameters
+    ----------
+    n_e: Quantity
+        Electron number density
+
+    Returns
+    -------
+    radius: Quantity
+        The Wigner-Seitz radius in meters
+
+    Raises
+    ------
+    TypeError
+        If argument is not a Quantity
+
+    UnitConversionError
+        If argument is in incorrect units
+
+    ValueError
+        If argument contains invalid values
+
+    UserWarning
+        If units are not provided and SI units are assumed
+
+    Notes
+    -----
+    The ideal chemical potential is given by [1]:
+
+    .. math::
+    \chi_a = I_{1/2}(\beta \mu_a^{ideal})
+    where \chi is the degeneracy parameter, I_{1/2} is the Fermi integral
+    with order 1/2, \beta is the inverse thermal energy \beta = 1/(k_B T),
+    and \mu_a^{ideal} is the ideal chemical potential.
```

Comment:
It would be really helpful to have this part explaining the mechanism by which the chemical potential is calculated. As I see it, you assume `I_{1/2} > \chi`, then try to find the closest value to a solution, right?

Reply:
So the equation is implicit, and I think it is essentially solving that damn Fermi integral for a particular case where the dense plasma is ideal. So then you have to numerically get the chemical potential since it is the argument of the Fermi integral.

Reply:
I've added a description for why we need to do this numerically.